### ISSUE TITLE
All internal links need to include full title text as well as the heading number

### ISSUE THREAD

**Issue by petervwyatt** [metanorma/metanorma-pdfa#11]
As many of our stakeholders require AT, the ISO style of only having a heading number is extremely inaccessible (and goes against WCAG)!

Please adjust to generate BOTH the heading number and the heading title text in both HTML and PDF. I tried setting `:xrefstyle: full` but it didn't seem to make a difference.

So this example should be "(see 3.11 PDF/A Requirements)" and the entire thing should be a longer hyperlink. This is then both fully descriptive when read aloud and also far more navigable for those with physical disabilities.

<img width="376" height="127" alt="Image" src="https://github.com/user-attachments/assets/c99c2b32-b5c6-4734-aaff-efc70f9c8335" />

**Comment by opoudjis** [comment]
Not only should xrefstyle be working, but PDFA use of it broke attachments in PDFs recently, because xrefstyle was being propagated to xref. I will need to investigate why this is happening, but my surmise is that, ISO has specific code to overrides these mechanisms (rejecting the "Clause" prefix before subclauses), and once again,  you are asking PDFA to do things that ISO is not supposed to do.

I will investigate this, and I may be able to tell metanorma-iso to not do the override of subclause numbers it does, if xrefstyle is set. 

But I am obligated to tell you that I may decline to implement some of PDFA's requests, as contrary to the intent of tastes. Tastes will not, under ANY circumstances, introduce code to override their base flavour's behaviour. Tastes are built to be config-only modifications of flavours, that modify presentation. They are the only way it is sustainable for Metanorma to address the presentation means of a large number of SDOs. We will not build a new flavour for every SDO that wants to use Metanorma. And there are clear limits in how open-ended I can or should be making ISO behaviour, to deal with PDFA requests.

**Comment by opoudjis** [comment]
ISO: subclause titles are not being recorded for use in full titles. I now am wondering whether I should parameterise the insertion or omission of "Clause" for subclause references in ISO, which could save code in JIS or BSI as well. I see BSI expects "Subclause" as the first word of a sentence, which is code not config.

**Comment by opoudjis** [comment]
Start with introducing "subclause" as the fallback label for subclauses, and recording title in Xref Anchors for subclauses, which I had not been doing.

**Comment by opoudjis** [comment]
Parameterising the display of "subclause" might be as easy as setting the anchor `elem` value to be `i18n.subclause`, and not defining it in ISO, but defining it in the downstream flavours.

The way `eref_locality_populate` is defined, `"#{locality} #{reference}"` makes the `locality` that prefixes the reference be `@i18n.locality[type] || type`. If i18n doesn't define "subclause", it will just use the string "subclause". For that reason, ISO should indeed define `i18n.subclause`—as "", not nil. That should still be safe.

**Comment by opoudjis** [comment]
Ah, I see you weren't asking for "Subclause 3.1.1 PDF/A Requirements", just "3.1.1 PDF/A Requirements", so I didn't need to introduce `i18n.subclause`, but someone else will ask for it eventually; BSI hints at it, but there it is optional, sentence-initial only.

This should now be working though, even in ISO.

**Comment by opoudjis** [comment]
```
:xrefstyle: full

== A
Reference: <<a>>

=== B

Content

[[a]]
==== My title

More content
```

now renders, natively in ISO (hence also in PDFA) as:

<img width="281" height="282" alt="Image" src="https://github.com/user-attachments/assets/f577f21c-3508-4cf9-bd8e-401827aaba4a" />

The one catch here is that we have hardcoded it to come out as _1.1.1, My title_ instead of _1.1.1 My title_

That is currently implemented as a method in isodoc, inherited by all of Metanorma:

```
    def anchor_xref_full(num, title)
      (!title.nil? && !title.empty?) or return nil
      l10n("#{num}<span class='fmt-comma'>,</span> #{title}")
    end
```

I can make that a configurable template, but I'm not going to do that for now, to get this out the door. If PDFA must get rid of the comma, I'll do so in a separate ticket.
